### PR TITLE
CIS-80 Remove + operator for pagination

### PR DESCRIPTION
**Pull request by b-onc**
...and use a set for paginations

### CODE REVIEW

File: Sources/Core/Client/Client+RxUnreadCount.swift [hunk 1]
```diff
@@ -32,13 +32,19 @@ public extension Reactive where Base == Client {
     /// - Note: 100 is the maximum unread count of messages.
     /// - Parameter channel: a channel.
     func channelUnreadCount(_ channel: Channel) -> Observable<ChannelUnreadCount> {
```
Comment (buh) [inline Sources/Core/Client/Client+RxUnreadCount.swift:34]: This method we decide to remove completely.

File: Sources/Client/Model/Pagination.swift [hunk 2]
```diff
@@ -8,6 +8,22 @@
...
+        first(where: { $0.limit != nil })?.limit
+    }
+    
+    var offset: Int? {
+        first(where: { $0.offset != nil })?.offset
+    }
+    
+    func encode(to encoder: Encoder) throws {
+        try forEach({ try $0.encode(to: encoder) })
+    }
```
Comment (buh) [inline Sources/Client/Model/Pagination.swift:24]: Why is that needed?

File: Sources/Core/Client/Channel/Channel+RxEvents.swift [hunk 3]
```diff
@@ -48,7 +48,18 @@ public extension Reactive where Base == Channel {
...
+                do {
+                    let response = try result.get()
+                    observer.onNext(response)
+                } catch {
+                    observer.onError(error)
+                }
+            }
+            
+            return Disposables.create { subscription.cancel() }
+        }
```
Comment (buh) [inline Sources/Core/Client/Channel/Channel+RxEvents.swift:62]: Why did you change this?

File: Sources/Core/Client/Client+RxWatcherCount.swift [hunk 4]
```diff
@@ -13,16 +13,17 @@ import RxSwift
...
-                    .startWith(channel.watcherCount)
-            })
+        .create { (observer) -> Disposable in
+            let subscription = channel.subscribeToWatcherCount { (result) in
+                do {
+                    let response = try result.get()
+                    observer.onNext(response)
+                } catch {
+                    observer.onError(error)
+                }
```
Comment (buh) [inline Sources/Core/Client/Client+RxWatcherCount.swift:23]: You do a lot of copy/paste this. Please check how it was implemented to avoid this repetition.

File: Sources/Client/Model/Pagination.swift (hunk 2 above)
Comment (b-onc) [reply Sources/Client/Model/Pagination.swift:24]: Otherwise `Set` encodes as `unkeyedContainer` which results in a runtime failure. We need it to encode in this format.

File: Sources/Core/Client/Channel/Channel+RxEvents.swift (hunk 3 above)
Comment (b-onc) [reply Sources/Core/Client/Channel/Channel+RxEvents.swift:62]: Because we don't have `channelUnreadCount` anymore.

File: Sources/Core/Client/Client+RxWatcherCount.swift (hunk 4 above)
Comment (b-onc) [reply Sources/Core/Client/Client+RxWatcherCount.swift:23]: Can you clarify? This is not copy/paste, this just relays the logic into LLC. This is not repetition either?

File: Sources/Core/Client/Channel/Channel+RxEvents.swift (hunk 3 above)
Comment (buh) [reply Sources/Core/Client/Channel/Channel+RxEvents.swift:62]: How it coupled with `Pagination`? :)

Comment (b-onc) [reply Sources/Core/Client/Channel/Channel+RxEvents.swift:62]: Changing pagination caused `channelUnreadCount` to not compile anymore. Instead of fixing it, I just removed it (via discussion) and removing it caused this to fail. So I refactored.

File: Sources/Core/Client/Client+RxWatcherCount.swift (hunk 4 above)
Comment (buh) [reply Sources/Core/Client/Client+RxWatcherCount.swift:23]: I think we can simplify that as it implemented to bind rx to completion block. Check `AutoCancellingSubscription` file.
If you can't, skip it.

Comment (b-onc) [reply Sources/Core/Client/Client+RxWatcherCount.swift:23]: You mean just return `channel.rx.watcherCount`?

Comment (buh) [reply Sources/Core/Client/Client+RxWatcherCount.swift:23]: No, all implementations should be on the `Client` (extensions) and on the `Channel` side is a short/handy way.

Comment (b-onc) [reply Sources/Core/Client/Client+RxWatcherCount.swift:23]: `Client.subscribeToWatcherCount(for channel` is LLC internal, so I can't use it here
